android/app/src/main/java/fi/fmi/mobileweather/BaseWidgetProvider.java

```diff
@@ -214,15 +226,11 @@ public void execute(String latlon) {
             try {
                 JSONObject result1 = future1.get();
                 JSONArray result2 = future2.get();
-                onPostExecute(result1, result2);
+                onPostExecute(result1, result2, main, pref);
             } catch (Exception e) {
                 Log.e("Download json", "Exception: " + e.getMessage());
-                showErrorView(
-                        context,
-                        pref,
-                        context.getResources().getString(R.string.update_failed),
-                        context.getResources().getString(R.string.check_internet_connection)
-                );
+                // NOTE: let's not show error view here, because connection problems with server
```

Comment:
Yes, it is not correct to show error view every time when forecast fetch fails. The data should be cached and used until it gets too old when fetch fails. For forecasts 24h is good limit.

I think that error view could be fixed, because there seems to already some logic for that. 

```
json = useNewOrStoredJsonObject(json, pref);
if (json == null) return;
```

Should show error view instead of just return?

Reply:
1x1 layout works fine in my Samsung device now.

Reply:
Do you mean an error view should be shown if last successful widget update happened more than 24h ago? 
Should that time limit even be shorter?

> Yes, it is not correct to show error view every time when forecast fetch fails. The data should be cached and used until it gets too old when fetch fails. For forecasts 24h is good limit.
> 
> I think that error view could be fixed, because there seems to be already some logic for that.

Reply:
Yes, show the error if there is more than 24h from the last forecast update.

If crisis message has not been updated that is not critical and no error should be shown.

Time limits should be discussed at some point, but I think that 24h is OK for forecasts. Warnings and observations can have shorter validity periods.